DEVELOPERS.md

```diff
@@ -269,74 +288,82 @@ pytest -m "workflow_test and not external_workflow_test"
+    pass
 
-Run the following command to view the available tasks:
 
-```shell
-task --list
-```
-```console
-* coverage:                            Run tests using pytest and generate a coverage report.
-* format:                              Format code using black.
-* format-check:                        Check code formatting using ruff.
-* lint:                                Run linting using ruff.
-* test:                                Run tests using pytest with the default flags defined in pyproject.toml.
-* test-all:                            Run all tests using pytest.
-* test-nf-core-pixelator:              Run the default nf-core/pixelator test profile with this version of pixelator.
-* test-web:                            Run web tests using pytest.
-* test-workflow:                       Run workflow tests using pytest.
-* test-workflow-external:              Run external workflow tests using pytest.
-* typecheck:                           Run type checking using mypy.
-* tests:update-report-test-data:       Update the report test data using the nf-core/pixelator test profile.
-* tests:update-web-test-data:          Create web test data.
-```
+@click.command()
+def my_command():
+    """My custom command"""
+    click.echo("Hello!")
 
-View more detailed documentation for a specific task:
 
-```shell
-task test-nf-core-pixelator --summary
+my_cli_plugin.add_command(my_command)
+
+
+if __name__ == "__main__":
+    sys.exit(my_cli_plugin())
 ```
-```console
-task: test-nf-core-pixelator
 
-Run the default nf-core/pixelator test profile with this version of pixelator.
+For pixelator to find this you need to define an `entrypoint https://setuptools.pypa.io/en/latest/userguide/entry_point.html#entry-points-for-plugins`_.
+Exactly how to do this depends on which packaging method you use to build your python package. In the link above you can find
+instructions for some common scenarios. The name you need use for your entry point is: `pixelator.cli_plugin`, so if you are using,
+`pyproject.toml` to configure your project it could look something like this:
 
-If the pipeline is not found in the directory `PIPELINE_SOURCE_DIR`, it will be cloned
-from the PixelgenTechnologies/nf-core-pixelator repository using the `PIPELINE_BRANCH` branch.
-By default the "pixelator-next" branch of nf-core-pixelator will be used.
+```{code-block} toml
+:caption: pyproject.toml
 
-If the `PIPELINE_SOURCE_DIR` exists this task will assume that the pipeline is already present and checked out
-in the right branch.
+[project.entry-points."pixelator.cli_plugin"]
+my_plugin = "my_plugin:my_cli_plugin"
+```
 
-Note that the current dev version of pixelator must be installed and available in the PATH.
+### Pixelator config plugins
 
-If the `RESUME` environment variable is set to "true", the pipeline will be resumed if it was previously run.
+Pixelator has a config object that contains information about assays loaded from pixelator.assays.
```

Comment:
I think that we shouldn't write yaml since using this mechanism you can load the config in what ever way you like, i.e. you could pull it from a database of a csv file. All you need to do is so make sure to implement the method as specified and then you can add to the `Config` instance.